ts/editor/NoteEditor.svelte

```diff
@@ -485,6 +486,44 @@ License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
         return false;
     }
 
+    // Signal editor UI state changes to add-ons
+
+    let editorState: EditorState = EditorState.Initial;
+    let lastEditorState: EditorState = editorState;
+
+    function getEditorState(
+        ioMaskEditorVisible: boolean,
```

Comment:
Essentially following the current render logic here. We'll probably be able to simplify this in the future (#2624)